### PR TITLE
Show list of Reports in the Report panel

### PR DESCRIPTION
**Pull request by Venkatesh-rengasamy**
### :nut_and_bolt: Description: What code changed, and why?

Add open report menu item in download-dropdown button and show list of download-reports with its status in the Report panel when you click the open report.

### :chains: Related Resources

#6191, #6193

### :+1: Definition of Done

Fetched the dowload-reports with its status and shown in the report panel(side panel) when you click open report menu. 

### :athletic_shoe: How to Build and Test the Change

inside hab,

rebuild the following components if you are getting 404 not found error for "custom_settings.js" api in network tab when page refresh or after login.
- rebuild components/automate-ui
- rebuild components/automate-gateway/
- rebuild components/automate-deployment/
- start_all_services

Rebuild the following components:-
- rebuild components/report-manager-minio-gateway/
- rebuild components/report-manager-service/
- rebuild components/compliance-service/
- rebuild components/automate-gateway/
- rebuild components/automate-cli/
- rebuild components/automate-deployment/
- start_all_services

create the config.toml file in /automate root folder. Add the below code in the file.
it enables the LCR configuration to test the new download flow.

[compliance.v1.sys.service]
upgrade_lcr = true

[global.v1.large_reporting]
enable_large_reporting = true

Then run these commands,
chef-automate config patch config.toml
chef_load_compliance_scans -D2 -N10 -M1 -T200

Login into automate-ui application, check the network tab -> "custom_settings.js" API returns -> large_reporting: { enable_large_reporting: true }. Click the compliance tab -> you can see the Report page -> you can see the download button near filter box -> click the download button -> you can see 'open report' menu is added in dropdown menu-> click either JSON or CSV format -> then click the open report menu -> it opens the side panel -> 
you can see the download report with its status details in the panel.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Comment by Venkatesh-rengasamy** [comment]
Screenshot from 2021-11-26 11-18-55
Screenshot from 2021-11-26 11-10-45


Screenshot from 2021-11-26 11-17-29

File: components/automate-ui/src/app/entities/download-reports/download-reports.service.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        this.reportList = responseData['data'];
+        this.checkReportStatus();
+      } else {
+        this.reportList = [];
+      }
+    });
+  }
+
+  fetchReportList() {
+    const url = `${REPORT_LIST_API_URL}/requests`;
```
Comment (Dmaddu) [inline components/automate-ui/src/app/entities/download-reports/download-reports.service.ts:35]: what is the reason behind defining the  url as 
download_report_list_url: '/api/v0/reportmanager',
and appending '/requests' here. 
why can't we define as below
download_report_list_url: '/api/v0/reportmanager/requests',

File: components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.html [hunk 2]
```diff
@@ -165,6 +166,46 @@ <h2 id="download-failed" class="display4" slot="title">Download failed.</h2>
...
+                  <chef-tooltip
+                  [attr.for]="'error-message-' + i"
+                  position="top"
+                  delay="0">{{ report.err_message }}</chef-tooltip>
+                </ng-container>
+              </tr>
+            </ng-container>
+          </table>
+          <ng-container *ngIf="downloadReportsService.reportList.length === 0">
+            <h4 class="display4" style="text-align: center;border-bottom: 5px solid #FFFFFF;">No reports are available to download.</h4>
```
[the remark this answers is not in the document]
Comment (Venkatesh-rengasamy) [reply components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.html]: removed inline css and created css class for this styles.

File: components/automate-ui/src/app/entities/download-reports/download-reports.service.ts (hunk 1 above)
Comment (Venkatesh-rengasamy) [reply components/automate-ui/src/app/entities/download-reports/download-reports.service.ts:35]: As we discussed, we are following common pattern in UI side. Created environment variable for this URL "'/api/v0/reportmanager" and '/requests' is used only in service file accordingly. In future we can reuse this variable with different endpoints.

**Review by vinay033 — APPROVED** [review]
LGTM

File: components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.html [hunk 3]
```diff
@@ -165,6 +166,46 @@ <h2 id="download-failed" class="display4" slot="title">Download failed.</h2>
...
+                  <chef-tooltip
+                  [attr.for]="'error-message-' + i"
+                  position="top"
+                  delay="0">{{ report.err_message }}</chef-tooltip>
+                </ng-container>
+              </tr>
+            </ng-container>
+          </table>
+          <ng-container *ngIf="downloadReportsService.reportList.length === 0">
+            <h4 class="display4" class="no-reports-message">No reports are available to download.</h4>
```
Comment (chaitali-mane) [inline components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.html:205]: Added class 2 times.

Comment (Venkatesh-rengasamy) [reply components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.html]: removed 2nd class attribute

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**